### PR TITLE
Refactor starknet_getEstimateFee + getEstimateFee error handling

### PR DESCRIPTION
**Pull request by dubzn**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: #397 

# What is the new behavior?
- starknet_estimateFee method now takes BlockId & BroadcastedTransaction parameters instead of BlockId and String.
- starknet_estimateFee now can throw BLOCK_NOT_FOUND, CONTRACT_NOT_FOUND and CONTRACT_ERROR errors via BeerusApiError

- Change starknet_estimateFee param from String to BroadcastedTransaction
- Add error handling for starknet_estimateFee (proposal for handler multiple types of errors), as described in the Starknet standard, estimateFee can throw the following [errors](https://github.com/starkware-libs/starknet-specs/blob/e0b76ed0d8d8eba405e182371f9edac8b2bcbc5a/api/starknet_api_openrpc.json#L483):
image

# Does this introduce a breaking change?

- [X] Yes
- [ ] No

Changing the method param of starknet_estimateFee
# Other information

### CODE REVIEW

File: crates/beerus-rpc/src/lib.rs [hunk 1]
```diff
@@ -372,32 +374,31 @@ impl BeerusApiServer for BeerusRpc {
...
             .await
-            .map_err(|e| Error::Call(CallError::Failed(anyhow::anyhow!(e.to_string()))))?;
-        Ok(estimate_fee)
+            .map_err(|e| {
+                let error_type: String = e.to_string();
+                match error_type.as_str() {
+                    BLOCK_NOT_FOUND => Error::from(BeerusApiError::BlockNotFound),
+                    CONTRACT_ERROR => Error::from(BeerusApiError::ContractError),
+                    CONTRACT_NOT_FOUND => Error::from(BeerusApiError::ContractNotFound),
+                    _ => Error::from(BeerusApiError::ContractError),
```
Comment (dubzn) [inline crates/beerus-rpc/src/lib.rs:399]: Let me know what you think of this, I could replicate it in methods that throw more than one error, but the way to do it is open to change.

File: crates/beerus-rpc/src/api.rs [hunk 2]
```diff
@@ -16,6 +17,10 @@ use starknet::{
         Transaction,
     },
 };
+use std::string::ToString;
+pub const BLOCK_NOT_FOUND: &str = "Block not found";
+pub const CONTRACT_NOT_FOUND: &str = "Contract not found";
+pub const CONTRACT_ERROR: &str = "Contract error";
```
Comment (dubzn) [inline crates/beerus-rpc/src/api.rs:23]: If we go this way I think it would be ideal to separate these constants in another file.

**Comment by glihm** [comment]
Synced with @dubzn to check this. Here a brief recap:

1. We have seen that lot's of work is done in `starknet-rs` about typings and codegen from the spec. So @dubzn will propose a rework of this PR based on those observations to stick to the generated code that have high chance to become the source of truth for all Rust-based implementation of starknet projects.

2. Even if the JSON-RPC spec of Starknet has a very defined set of errors, we have a situation. The implementation **MUST** deal with invalid value. For instance, if the error code sent back by the node in the response is `500000`, we can't map that to an actual value of the specification. For this reason, dub will explore a way to correctly handle this.

3. Finally, for ETH endpoints, we will remove any abstraction possible to work with helios types as much as possible, to ensure that future users of Beerus (that may use helios today) can find the exact same API.

4. Dub will probably integrate this PR in the other PR more generic with all endpoints.
   @dubzn will focus on only on Starknet endpoints.

@phklive @drspacemn this is the debrief from @dubzn and I after syncing today on this. :+1:

**Comment by phklive** [comment]
Okay great, let's see what that translates in in terms of code and let's go 👍

**Review by drspacemn — APPROVED** [review]
lgtm

File: crates/beerus-rpc/src/api.rs (hunk 2 above)
Comment (drspacemn) [reply crates/beerus-rpc/src/api.rs:27]: good for now, can separate at a later date

File: crates/beerus-rpc/src/lib.rs (hunk 1 above)
Comment (drspacemn) [reply crates/beerus-rpc/src/lib.rs:689]: good for now

**Pull request merged**